src/app/core/workbox/workbox.component.ts

```diff
@@ -221,16 +303,35 @@ export class WorkBoxComponent extends DirectiveBase implements OnInit, AfterView
     }
 
     /**
-     * Removes all tabs except one
+     * Removes all tabs or all other tabs (if tab is passed as an argument)
      */
-    private removeOtherTabs(tab) {
-        this.workbox.closeOtherTabs(tab);
-    }
+    private removeAllTabs(tab?: TabData<any>) {
+
+        const component = tab ? this.getTabComponent(tab).tabComponent : null;
+
+        const hasDirtyTabs = this.tabComponents.find((tabComponent) => {
+            return tabComponent.appIsDirty && (tabComponent !== component);
+        });
+
+        const modalTitle = `Close ${tab ? "other" : "all"} tabs`;
+
+        if (hasDirtyTabs) {
+
+            // If there are apps that are Dirty show modal
+            const modal = this.modal.confirm({
+                title: modalTitle,
+                content: `Some documents are modified.<br>
```

Comment:
It would be great if we could rely on implicitly added `\n` symbols for new lines, since we are going to move these messages to xlf translation files at some point, and they shouldn't contain HTML.